libvast/src/table_slice.cpp

```diff
@@ -229,21 +231,10 @@ table_slice table_slice::unshare() const noexcept {
 
 // TODO: Dispatch to optimized implementations if the encodings are the same.
 bool operator==(const table_slice& lhs, const table_slice& rhs) noexcept {
-  // Check whether the slices point to the same chunk of data.
-  if (lhs.chunk_ == rhs.chunk_)
+  if (!lhs.chunk_ && !rhs.chunk_)
     return true;
-  // Check whether the slices have different sizes or layouts.
-  if (lhs.rows() != rhs.rows() || lhs.columns() != rhs.columns()
-      || lhs.layout() != lhs.layout())
-    return false;
-  // Check whether the slices contain different data.
-  auto flat_layout = flatten(caf::get<record_type>(lhs.layout()));
-  for (size_t row = 0; row < lhs.rows(); ++row)
-    for (size_t col = 0; col < flat_layout.num_fields(); ++col)
-      if (lhs.at(row, col, flat_layout.field(col).type)
-          != rhs.at(row, col, flat_layout.field(col).type))
-        return false;
-  return true;
+  constexpr auto check_metadata = true;
+  return to_record_batch(lhs)->Equals(*to_record_batch(rhs), check_metadata);
```

Comment:
We should probably still use the old logic for `msgpack`?

Reply:
I did consider that, although there's only very little of that remaining. We switched the default so long ago that I think we should not maintain additional code for it, and we are actively warning on startup if partitions with a partition version of 0 exist.

Reply:
I added a comment explaining my reasoning.

Reply:
That reasoning doesn't fully convince me. We know of old deployments that still have data from more than a year ago. Just because we don't use this function in prod right now doesn't mean we won't in the future, and in theory there could be plugins using it too.

Reply:
I added it back.